File: Objective-C/TripServiceKata/TripServiceKata/TripService.m
#import "TripService.h"
#import "User.h"
#import "UserSession.h"
#import "TripDAO.h"
#import "UserNotLoggedInException.h"

@implementation TripService

- (NSArray*)GetTripsByUser:(User*)user
{
    NSArray* tripList = [NSArray array];
    User* loggedUser = [[UserSession sharedInstance] getLoggedUser];
    
    BOOL isFriend = NO;
    
    if(loggedUser != nil)
    {
        for (User* friend in [user getFriends])
        {
            if(friend == loggedUser)
            {
                isFriend = YES;
                break;
            }
        }
        if(isFriend)
        {
            tripList = [TripDAO findTripsByUser:user];
        }
        
        return tripList;
    }
    else
    {
        [UserNotLoggedInException raise:@"Unit Test Exception" format:@""];
    }
    return  nil;
}

@end

File: Swift 4.x/TripServiceKata/Sources/TripServiceKata/TripService.swift
import Foundation

class TripService
{
    func getTripsByUser(_ user:User) throws -> [Trip]?
    {
        var tripList:[Trip]? = nil
        let loggedUser = try! UserSession.sharedInstance.getLoggedUser()
        
        var isFriend = false
        
        if let loggedUser = loggedUser {
            for friend in user.getFriends() {
                if friend == loggedUser {
                    isFriend = true
                    break
                }
            }
            if isFriend {
                tripList = try! TripDAO.findTripsByUser(user)
            }
            return tripList
        }
        else {
            throw TripServiceErrorType.userNotLoggedIn
        }
    }
}
